quickwit/quickwit-indexing/src/source/gcp_pubsub_source.rs

```diff
@@ -219,10 +228,25 @@ impl Source for GcpPubSubSource {
 
     async fn suggest_truncate(
         &mut self,
-        _checkpoint: SourceCheckpoint,
+        checkpoint: SourceCheckpoint,
         _ctx: &ActorContext<SourceActor>,
     ) -> anyhow::Result<()> {
-        // TODO: add ack of ids
+        if let Some(Position::Offset(offset_str)) = checkpoint.position_for_partition(&self.partition_id) {
+            let commited_until = offset_str.parse::<Ulid>()?;
+            while let Some((position, messages)) = self.in_flight_batches.front() {
+                if position > &commited_until {
+                    break
+                }
+                // TODO, we could avoid this clone if we pop and push back if fail to ack ?
```

Comment:
Yes, indeed, the handler of the `SuggestTruncate` message only logs an error:
```rust
#[async_trait]
impl Handler<SuggestTruncate> for SourceActor {
    type Reply = ();

    async fn handle(
        &mut self,
        suggest_truncate: SuggestTruncate,
        ctx: &SourceContext,
    ) -> Result<(), ActorExitStatus> {
        let SuggestTruncate(checkpoint) = suggest_truncate;
        if let Err(err) = self.source.suggest_truncate(checkpoint, ctx).await {
            // Failing to process suggest truncate does not
            // kill the source nor the indexing pipeline, but we log the error.
            error!(err=?err, "suggest-truncate-error");
        }
        Ok(())
    }
}
```

Reply:
Then on ack failure, should we also extend the deadline? This can also fail, I guess. If we are never able to ack, we have to gc the message IDs at some point. Whatever we do, we just get a better shade of at-least-once but the code complexity increases dramatically. I suggest we ignore ack errors for now.

Does the client handle retries automatically for us?

Reply:
> Does the client handle retries automatically for us?

it does with the default settings. (in the ack methods I cant set the retry).
```
impl Default for RetrySetting {
    fn default() -> Self {
        Self {
            from_millis: 10,
            max_delay: Some(Duration::from_secs(1)),
            factor: 1u64,
            take: 5,
            codes: vec![Code::Unavailable, Code::Unknown, Code::Aborted],
        }
    }
}

```

Reply:
Awesome. We're in good shape if we apply this retry policy for all the GCP API calls.

Reply:
I think it is by default, so we should be good. Some `fn` take an optional Retry, so we could tune it later (I guess default should be fine for now)